### PR TITLE
Support load_state_dict func for nn.MultiheadAttention

### PR DESCRIPTION
**Pull request by zhangguanheng66**
Add _load_from_state_dict func in nn.MultiheadAttention func. JIT currently doesn't support `hasattr` func. We choose to apply load_state_dict func for the backward compatibility. Current users should not be affected by the change. Relevant PR #24204 

The changes have been tested to load a MultiheadAttention model trained by PyTorch 1.1. If users have an old MultiheadAttention model, please consider the following script to run inference.

```
import torch
import torch.nn as nn

model = torch.load("model_old.pt")  # Load an old model
state_dict = model.state_dict()
new_model = nn.MultiheadAttention(64, 4)  # A model with the new version of MultiheadAttention 
new_model.load_state_dict(state_dict)
```

### CODE REVIEW

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -756,7 +762,7 @@ def forward(self, query, key, value, key_padding_mask=None,
         - attn_output_weights: :math:`(N, L, S)` where N is the batch size,
           L is the target sequence length, S is the source sequence length.
         """
-        if hasattr(self, '_qkv_same_embed_dim') and self._qkv_same_embed_dim is False:
+        if not self._qkv_same_embed_dim:
```
Comment (soumith) [inline torch/nn/modules/activation.py]: do you mean `if self._qkv_same_embed_dim is None`?
`if not` can be ambiguous, if it's an integer of value `0`, or `False` or `None`

File: torch/nn/modules/activation.py [hunk 2]
```diff
@@ -768,11 +774,6 @@ def forward(self, query, key, value, key_padding_mask=None,
                 q_proj_weight=self.q_proj_weight, k_proj_weight=self.k_proj_weight,
                 v_proj_weight=self.v_proj_weight)
         else:
-            if not hasattr(self, '_qkv_same_embed_dim'):
```
Comment (soumith) [inline torch/nn/modules/activation.py]: the PR description of "user has to do this" cannot be known by the user magically.

When user does `model = torch.load('model.pt')`, they have no idea that they are supposed to create a new MultiHeadAttention and load the state dict into it.
This is not a good fix, just to make the JIT happy.

**Review by soumith — CHANGES_REQUESTED** [review]

File: torch/nn/modules/activation.py (hunk 2 above)
Comment (zhangguanheng66) [reply torch/nn/modules/activation.py]: With the current code, users don't need to worry the BC breaking. We will communicate with the JIT team.

File: torch/nn/modules/activation.py [hunk 3]
```diff
@@ -726,6 +726,12 @@ def _reset_parameters(self):
         if self.bias_v is not None:
             xavier_normal_(self.bias_v)
 
+    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
+                              missing_keys, unexpected_keys, error_msgs):
+        super(MultiheadAttention, self)._load_from_state_dict(
```
Comment (driazati) [inline torch/nn/modules/activation.py]: Just forwarding a call to the `super` shouldn't do anything extra, shouldn't this be adding `_qkv_same_embed_dim` to the `state_dict` with a default value if it's not present in `state_dict` already (meaning the state dict is from an older model) so the user doesn't have to do anything?

Comment (zhangguanheng66) [reply torch/nn/modules/activation.py]: No. _qkv_same_embed_dim should not be in state_dict. state_dict should only have learnable parameter.
In this case, state_dict tracks in_proj_weight/in_proj_bias/out_proj.weight/out_proj.bias.

**Pull request closed without merging**